process/pty.go

```diff
@@ -1,4 +1,4 @@
-// +build !windows
+// +build linux,netbsd,freebsd,dragonfly darwin,386 darwin,amd64 openbsd,386 openbsd,amd64
```

Comment:
Do you have any links/reference/anecdotes explaining why this is the right list of architectures? Same question for `process/pty_unsupported.go`. It would probably be helpful for future generations to understand, and give people confidence to change it if/when required.

Reply:
Thanks, I added some comments.